vtex/go-clients
# Add ETag to SaveFileOptions

#### What is the purpose of this pull request?
A user of the VBase client should use this field to pass along the `eTag` (from a get), so VBase can if-match it.

#### What problem is this solving?
Unable to use `if-match` feature.

#### How should this be manually tested?
I tested by releasing a tag `go-clients@v0.1.0-etag`, but then I noticed that this doesn't work here and I actually released a tag. I installed in `Courier` so I can test it, but I already learnt that I actually should've used `go mod edit -replace github.com/vtex/go-clients=<local path>`.

Anyway, my test to check whether the `if-match` was working was: After I got an event from SQS and got a batch and eTag from VBase's client, I copied the event and called VBase's client again to get the batch and eTag (it should've been the same), then I saved the copy. Then, it continued to save the real event with the first eTag and it failed with an error of `412` as expected.

#### Types of changes
